tests/e2e/support/page/files/version.ts

```diff
@@ -28,4 +39,33 @@ export class VersionPage {
     const elements = this.fileVersionSidebarXPath
     return await elements.count()
   }
+
+  async restoreOlderVersion({ files, folder }: { files: File[]; folder?: string }) {
+    const { page } = this.actor
+    const { dir: resourceDir, base: resourceBase } = path.parse(folder)
+    const fileName = files.map((file) => path.basename(file.name))
+    await filesCta.navigateToFolder({ page: page, path: folder })
+    await filesCta.sidebar.open({ page: page, resource: fileName[0] })
+    await filesCta.sidebar.openPanel({ page: page, name: 'versions' })
+    const fileSize = await this.fileVersionSizeXPath.textContent()
+    VersionPage.olderVersionSize = fileSize.trim()
+    await Promise.all([
+      page.waitForResponse(
+        (resp) =>
+          resp.url().includes('/v/') && resp.status() === 204 && resp.request().method() === 'COPY'
+      ),
+      await this.fileVersionRevertButtonXPath.click()
+    ])
+    // Files details page does not update after restore button is clicked
+    await page.reload()
```

Comment:
I tried running the tests without it. with the backend of oCis and oC10 the tests passed

Reply:
I used the page reload because there is issue on web ( https://github.com/owncloud/web/issues/6361 ), as we are comparing size from UI reload is necessary. May be the test have been passed for you because the size was same on both versions.

Reply:
That is dangerous if we just make this pass, it will cover up the bug.
But I think we also don't want to introduce something like the expected to fail file here because then we would have to mark the whole user-story as "expected to fail" and by that miss all the other checks it does.

@ScharfViktor what to do?

Reply:
Options that I can see:
- have tests that actually don't show the expected behavior but the current behavior, see background [in oc docs](https://doc.owncloud.com/server/10.9/developer_manual/testing/acceptance-tests.html#writing-scenarios-for-bugs) or [this blogpost](https://dev.to/jankaritech/should-you-write-acceptance-tests-for-bugs-2bfm)
- skip any action that has bugs in the e2e tests and first create a unit-test that shows the issue

Reply:
In the general case, when a user story has steps that currently do not pass because the system-under-test has a bug, then what are we going to do. We need to decide that in general. @individual-it gave some options above. Here are some more that I can think of:

- put the failing step(s) at the end of the user story (as far as possible, make it so that all the other good steps do execute before the failure happens, and
- have something that records the actual step that fails, and the CI can parse the test result and validate that the test really fails at the "expected" test step

Then if a change happens to the system-under-test, and the user story starts passing, or fails earlier (regression), or fails later (the direct bug is fixed but some later step is still a problem) then CI will fail and the developer will be aware of a change in behavior, and can look to see if the behavior change is a regression or improvement.

If the future direction is that the nightwatch test suite is going to reduce (or go away), and the Playwright e2e test suite is going to become the way that we provide some broad "QA anti-regression acceptance end-2-end (insert your favorite word for these kind of tests)" test coverage that helps everyone (QA, developers, users...) have confidence that the functionality is at least as good as previously, and that the functionality is improving, then we need to decide what to do when there are tests that cover functionality that currently does not 100% meet the requirements.

Reply:
I have no quick ideas on how to solve this problem. I created a issue https://github.com/owncloud/web/issues/6385 to don't lock this PR